src/renderer/examples/main.rs

```diff
@@ -0,0 +1,234 @@
+
+extern crate cgmath;
+extern crate gfx;
+extern crate gfx_device_gl;
+extern crate gfx_window_glutin;
+extern crate glutin;
+extern crate genmesh;
+extern crate amethyst_renderer;
+extern crate rand;
+
+use std::time::SystemTime;
+use std::collections::HashMap;
+use rand::Rng;
+
+use gfx::{Device};
+use gfx::traits::FactoryExt;
+
+use cgmath::{Point3, Vector3, Matrix4, EuclideanVector};
+use cgmath::{Transform, AffineMatrix3};
+use genmesh::generators::SphereUV;
+use genmesh::{Triangulate, MapToVertices, Vertices};
+
+use amethyst_renderer::VertexPosNormal as Vertex;
+use amethyst_renderer::{ColorFormat, DepthFormat};
+
+fn build_sphere() -> Vec<Vertex> {
+    SphereUV::new(32, 32)
+        .vertex(|(x, y, z)| Vertex{
+            pos: [x, y, z],
+            normal: Vector3::new(x, y, z).normalize().into()
+        })
+        .triangulate()
+        .vertices()
+        .collect()
+}
+
+fn main() {
+    let builder = glutin::WindowBuilder::new()
+        .with_title("Amethyst Renderer Demo".to_string())
+        .with_dimensions(800, 600)
+        .with_vsync();
+
+    let (window, mut device, mut factory, main_color, main_depth) =
+        gfx_window_glutin::init::<ColorFormat, DepthFormat>(builder);
+    let combuf = factory.create_command_buffer();
+
+    let sphere = build_sphere();
+    let buffer = factory.create_vertex_buffer(&sphere);
```

Comment:
you can use `let (buffer, slice) = factory.create_vertex_buffer_with_slice(&sphere, ());`